pkg/cfn/manager/create_tasks.go

```diff
@@ -133,15 +134,17 @@ func (c *StackCollection) NewTasksToCreateIAMServiceAccounts(serviceAccounts []*
 			saTasks.Append(&kubernetesTask{
 				info:       fmt.Sprintf("create serviceaccount %q", sa.NameString()),
 				kubernetes: clientSetGetter,
-				call: func(clientSet kubernetes.Interface) error {
+				objectMeta: sa.ClusterIAMMeta.AsObjectMeta(),
+				call: func(clientSet kubernetes.Interface, objectMeta v1.ObjectMeta) error {
 					sa.SetAnnotations()
 					if sa.Labels == nil {
 						sa.Labels = make(map[string]string)
 					}
 
 					sa.Labels[managedByKubernetesLabelKey] = managedByKubernetesLabelValue
-					if err := kubernetes.MaybeCreateServiceAccountOrUpdateMetadata(clientSet, sa.ClusterIAMMeta.AsObjectMeta()); err != nil {
-						return errors.Wrapf(err, "failed to create service account %s", sa.NameString())
+					objectMeta = sa.AsObjectMeta()
+					if err := kubernetes.MaybeCreateServiceAccountOrUpdateMetadata(clientSet, objectMeta); err != nil {
+						return errors.Wrapf(err, "failed to create service account %s/%s", objectMeta.GetNamespace(), objectMeta.GetName())
```

Comment:
Slightly confused by this, we set the `objectMeta`, which then gets passed in as an argument to the func (from my understand), but then we override the provided by getting the meta again? Should we just be relying on the object meta input and not re-fetching it?

Reply:
The annotations are missing in the function argument object meta because the sa object status does not have the role arn. As the role is created only in the predecessor task of Kubernetes sa create task.

Reply:
what is the purpose of passing `objectMeta` into the call func, if it is not useful to us?

Reply:
I think it make sense to just update annotations on objectMeta than overriding whole objectMeta

Reply:
> I think it make sense to just update annotations on objectMeta than overriding whole objectMeta

I am not sure what you mean here; where are we updating just the annotations? Is that line `139`?

sorry i may be getting confused 😅 .

this is what i see happening:
1. we are passing in `objectMeta` as a param to the `callFunc` on line `138`
1. this `objectMeta` is not used, it is overwritten on line `145`
1. we pass the new `objectMeta` on line `146`

If I remove the `objectMeta` coming in on `138`, create an SA, nothing changes: the SA is created with the correct `objectMeta`.

So why have the param in the `callFunc` when it appears to serve no purpose?

Reply:
>  Is that line 139?

Yes, you are correct.
What you are seeing happening is also correct.

> So why have the param in the callFunc when it appears to serve no purpose?

Kubernetes task needs to have object meta passed in.

I have updated the code to not override, but just update objectMeta with annotation and labels from SA.

Reply:
sounds good 😃